beacon_node/http_api/src/lib.rs

```diff
@@ -1933,6 +1933,77 @@ pub fn serve<T: BeaconChainTypes>(
             })
         });
 
+    /*
+     * builder/states
+     */
+
+    let builder_states_path = eth_v1
+        .and(warp::path("builder"))
+        .and(warp::path("states"))
+        .and(chain_filter.clone());
+
+    // GET builder/states/{state_id}/expected_withdrawals
+    let get_expected_withdrawals = builder_states_path
+        .clone()
+        .and(warp::path::param::<StateId>())
+        .and(warp::query::<api_types::ExpectedWithdrawalsQuery>())
+        .and(warp::path("expected_withdrawals"))
+        .and(warp::path::end())
+        .and(warp::header::optional::<api_types::Accept>("accept"))
+        .and_then(
+            |chain: Arc<BeaconChain<T>>,
+             state_id: StateId,
+             query: api_types::ExpectedWithdrawalsQuery,
+             accept_header: Option<api_types::Accept>| {
+                blocking_response_task(move || {
+                    let proposal_slot = query.proposal_slot.unwrap_or(Slot::new(1));
+                    let (state, execution_optimistic, finalized) = match state_id.state(&chain) {
+                        Ok(state) => state,
+                        Err(_) => {
+                            return Err(warp_utils::reject::custom_not_found(
+                                "State not found".to_string(),
+                            ))
+                        }
+                    };
+                    let forkchoice_update_parameters = chain
+                        .canonical_head
+                        .cached_head()
+                        .forkchoice_update_parameters();
+
+                    let withdrawals = match chain.get_expected_withdrawals(
+                        &forkchoice_update_parameters,
+                        state.slot() + proposal_slot,
+                    ) {
+                        Ok(withdrawals) => withdrawals,
+                        Err(_) => {
+                            return Err(warp_utils::reject::custom_bad_request(
+                                "The specified state is not a capella state.".to_string(),
```

Comment:
I think this error may be incorrect as an error from `chain.get_expected_withdrawals` does not indicate that the state isn't a capella state. It seems like errors from this function indicates an expected condition and should probably return 500. 

Although I think we should validate that the `proposal_slot` is a slot after the capella fork, and return an error if the `proposal_slot` is pre-capella.